src/Components/Modules/Dvar.cpp

```
@@ -97,32 +97,26 @@ namespace Components
 	}
 	void Dvar::Var::set(int integer)
 	{
-		if (this->dvar && this->dvar->name)
+		if (this->dvar && this->dvar->type == Game::dvar_type::DVAR_TYPE_INT)
 		{
-			Game::Dvar_SetCommand(this->dvar->name, Utils::String::VA("%i", integer));
+			this->dvar->current.integer = integer;
+			this->dvar->latched.integer = integer;
 		}
 	}
 	void Dvar::Var::set(float value)
 	{
-		if (this->dvar && this->dvar->name)
-		{
-			Game::Dvar_SetCommand(this->dvar->name, Utils::String::VA("%f", value));
-		}
-	}
-
-	void Dvar::Var::setRaw(int integer)
-	{
-		if (this->dvar)
+		if (this->dvar && this->dvar->type == Game::dvar_type::DVAR_TYPE_FLOAT)
 		{
-			this->dvar->current.integer = integer;
+			this->dvar->current.value = value;
+			this->dvar->latched.value = value;
 		}
 	}
-
-	void Dvar::Var::setRaw(float value)
+	void Dvar::Var::set(bool enabled)
 	{
-		if (this->dvar)
+		if (this->dvar && this->dvar->type == Game::dvar_type::DVAR_TYPE_BOOL)
```

i would add an assert here (before this line) that checks for the dvar type to not silently fail in debug mode. 
```cpp
assert(this->dvar->type == Game::DVAR_TYPE_BOOL);
```